src/components/shoppingCart/cartdetail/CartDetail.jsx

```diff
@@ -3,11 +3,54 @@ import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 import Button from "react-bootstrap/Button";
-import { MOCK_DATABASE } from "../../common/MockDatabase";
+import "./CartDetail.scss";
+import {
+  deleteItemFromCartAction,
+  increaseProductQuantity,
+  decreaseProductQuantity,
+  sumTotalPrice,
+  addOneMoreItem,
+  removeOneMoreItem,
+} from "../../redux/actions";
+import { useDispatch, useSelector } from "react-redux";
 
-const { title, images, brand, price } = MOCK_DATABASE[0];
+const CartDetail = (props) => {
+  const dispatch = useDispatch();
+  const productsData = useSelector((state) => state.addOrDeleteProductData);
+  // Getting all the total prices from individual items and add them together
+  const itemsTotalPricesArray = useSelector((state) => state.getTotalPrice);
+  const allItemsTotalPrice = itemsTotalPricesArray.reduce((cur, acc) => {
+    return cur + acc;
+  }, 0);
+
+  // Get total price from one item
+  React.useEffect(() => {
+    dispatch(sumTotalPrice(props.info.price, props.info.quantity));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // This function deletes thw whole product from the shopping cart and subtract the corresponding price from the total price
+  const deleteItemfromCartData = () => {
+    const copyOfData = productsData.slice();
+    for (let i = 0; i < copyOfData.length; i++) {
+      const currentItemId = props.info.id;
+      if (copyOfData[i].id === currentItemId) {
+        if (i > -1) {
+          copyOfData.splice(i, 1);
+        }
+        dispatch(
+          removeOneMoreItem(
+            allItemsTotalPrice,
+            props.info.price * props.info.quantity
+          )
+        );
+        dispatch(deleteItemFromCartAction(copyOfData));
```

Comment:
if i'm understanding your previous comment, it doesn't make sense that an action called `deleteItemFromCartAction` simply updates the cart with the information that it takes. if i saw such an action, i would expect:

1. deleteItem takes one item
2. it deletes the current item from the cart

right now, your action here ism ore like "update item"

or it seems to me that "removeonemoreitem" is actually the same as deleteitemfromcartaction?

i'm actually not sure i understand this code, but how i would envision a cart being implemented is that:
* you have a cart object
* you have operations on the cart object (add, delete)
* all you have to do in the components is add an item (price, quantity) to the cart
* use a memoized selector to get the total and so on: https://redux.js.org/recipes/computing-derived-data

Reply:
This function deletes the whole item with its quantity and then subtract the total price of the deleted item from the total price of all items